cmd/completion.go

```diff
@@ -0,0 +1,194 @@
+// Copyright 2020 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"bytes"
+	"io"
+	"os"
+
+	"github.com/pingcap/errors"
+	"github.com/spf13/cobra"
+)
+
+func newCompletionCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "completion SHELL",
+		Short: "Output shell completion code for the specified shell (bash or zsh)",
+		Example: `  # Installing bash completion on macOS using homebrew
+  ## If running Bash 3.2 included with macOS
+  brew install bash-completion
+  ## or, if running Bash 4.1+
+  brew install bash-completion@2
+
+
+  # Installing bash completion on Linux
+  ## If bash-completion is not installed on Linux, please install the 'bash-completion' package
+  ## via your distribution's package manager.
+  ## Load the tiup completion code for bash into the current shell
+  source <(tiup completion bash)
+  ## Write bash completion code to a file and source if from .bash_profile
+  tiup completion bash > ~/.completion.bash.inc
+  printf "
+  # tiup shell completion
+  source '$HOME/.completion.bash.inc'
+  " >> $HOME/.bash_profile
+  source $HOME/.bash_profile
+
+
+  # Load the tiup completion code for zsh[1] into the current shell
+  source <(tiup completion zsh)
+  # Set the tiup completion code for zsh[1] to autoload on startup
+  tiup completion zsh > "${fpath[1]}/_tiup"`,
+		SilenceUsage:  true,
+		SilenceErrors: true,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if len(args) == 1 {
+				switch args[0] {
+				case "bash":
+					return cmd.Parent().GenBashCompletion(os.Stdout)
+				case "zsh":
+					return genCompletionZsh(os.Stdout, cmd.Parent())
+				default:
+					return errors.New("Unsupported shell type " + args[0])
```

Comment:
The error should begin with the lower-case according to Golang idiomatic.

Reply:
Done